### PR TITLE
[EBPF] Add extra stats to KMT jobs

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR increases our visibility into KMT pipelines and CI jobs, with the following changes:

- Sends environment variables such as job architecture, test component and similar variables as CI job tags for easier classification of jobs
- Detects test results and generates several tags that can be used to detect whether the job has failed due to infrastructure issues or due to actual test failures.
- Classifies the stage of each job for improved classification  
- Adds a `kmt.tag-ci-job` that centralizes the generation of tasks
- Adds a branch tag to the AWS instances, for easier tracking of per-PR costs. 

### Motivation

Improve visibility into KMT, specially around infrastructure failures, to generate data that will be used to inform development decisions.

### Possible Drawbacks / Trade-offs

The `datadog-ci` standalone binary is not available in ARM64 runners, so those jobs will not get tagged correctly. Right now it seems that the complexity of adding tags to those jobs is greater than the benefit we'd get from it (specially as it seems we don't have a significant difference of outcomes between x64/arm64 jobs) so, for now, the CI will not execute the tagging task if the `datadog-ci` binary is not available.

### CODE REVIEW

File: .gitlab/kernel_matrix_testing/common.yml [hunk 1]
```diff
@@ -104,7 +118,7 @@
     INFRA_ENV: "aws/agent-qa"
     PIPELINE_ID: $CI_PIPELINE_ID
     TEAM: "ebpf-platform"
-    RESOURCE_TAGS: "instance-type:${INSTANCE_TYPE},arch:${ARCH},test-component:${TEST_COMPONENT}"
+    RESOURCE_TAGS: "instance-type:${INSTANCE_TYPE},arch:${ARCH},test-component:${TEST_COMPONENT},branch:${CI_COMMIT_REF_NAME}"
```
Comment (usamasaqib) [inline .gitlab/kernel_matrix_testing/common.yml:121]: nit
```suggestion
    RESOURCE_TAGS: "instance-type:${INSTANCE_TYPE},arch:${ARCH},test-component:${TEST_COMPONENT},git-branch:${CI_COMMIT_REF_NAME}"
```

File: tasks/kmt.py [hunk 2]
```diff
@@ -1742,3 +1743,79 @@ def _color_result(result: tuple[int, int, int]) -> str:
...
+    """Add extra tags to the CI job"""
+    tags: dict[str, str] = {}
+
+    # Retrieve tags from environment variables, with a renaming for convenience
+    environment_vars_to_tags = {
+        "ARCH": "arch",
+        "TEST_COMPONENT": "component",
+        "TAG": "platform",
+        "TEST_SET": "test_set",
+        "MICRO_VM_IP": "microvm_ip",
```
Comment (usamasaqib) [inline tasks/kmt.py:1759]: Why is the `microvm-ip` useful as a tag?

**Review by usamasaqib — APPROVED** [review]

File: tasks/kmt.py (hunk 2 above)
Comment (gjulianm) [reply tasks/kmt.py:1759]: I added it so we can count the number of VMs per instance, and also to be able to easily link jobs that are running in the same VM. Not super-useful but assuming the cost of these tags is negligible I thought it was worth it.

**Review by YoannGh — APPROVED** [review]

**Pull request merged**